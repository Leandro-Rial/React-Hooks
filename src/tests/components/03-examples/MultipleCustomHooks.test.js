import { shallow } from "enzyme";
import MultipleCustomHooks from "../../../components/03-examples/MultipleCustomHooks";
import useCounter from "../../../components/hooks/useCounter";
import useFetch from "../../../components/hooks/useFetch";
jest.mock("../../../components/hooks/useFetch")
jest.mock("../../../components/hooks/useCounter")

describe('Pruebas en MultipleCustomHooks', () => {

    beforeEach( () =>  {
 
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });
 
    });

    test('debe de mostrar <MultipleCustomHooks /> correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })
        
        const wrapper = shallow(<MultipleCustomHooks />)

        expect(wrapper).toMatchSnapshot()

    });
    
    test('debe de mostrar la informacion', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Leandro',
                quote: "Helouda"
            }],
            loading: false,
            error: null
        });
        
        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-4').text().trim()).toBe('Helouda');
        expect(wrapper.find('footer').text().trim()).toBe('Leandro');

    });
});