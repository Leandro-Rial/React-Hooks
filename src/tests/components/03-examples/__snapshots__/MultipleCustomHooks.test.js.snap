// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultipleCustomHooks debe de mostrar <MultipleCustomHooks /> correctamente 1`] = `
<div>
  <h1>
    BreackingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
